flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/PartitionRequestClientFactory.java

```diff
@@ -103,6 +105,10 @@ NettyPartitionRequestClient createPartitionRequestClient(ConnectionID connection
             } else {
                 try {
                     client = clientFuture.get();
+                    if (client.canBeDisposed()) {
+                        client.closeConnection();
+                        continue;
+                    }
```

Comment:
Isn't this duplicating the following?
```
            if (client.incrementReferenceCounter()) {
                return client;
            } else {
                destroyPartitionRequestClient(connectionId, client);
            }
```

Reply:
I'd suggest to keep the latter, as it decouples from the `if-else`/`try-catch` branches.

Moreover, I'd suggest to rename `incrementReferenceCounter` to `validateClientAndIncrementReferenceCounter`.